six/modules/c++/six.sicd/source/Grid.cpp

```diff
@@ -84,4 +471,831 @@ bool Grid::operator==(const Grid& rhs) const
+        collectionInformation.radarMode == RadarModeType::SPOTLIGHT &&
+        Init::isUndefined<Poly2D>(timeCOAPoly))
+    {
+        timeCOAPoly = Poly2D(0, 0);
+        timeCOAPoly[0][0] = scpcoa.scpTime;
+    }
+
+    row->fillDerivedFields(imageData);
+    col->fillDerivedFields(imageData);
+}
+
+void Grid::fillDerivedFields(const RMA& rma, const Vector3& scp,
+        const PolyXYZ& arpPoly)
+{
+    if (rma.rmat.get())
+    {
+        fillDerivedFields(*rma.rmat, scp);
+    }
+    else if (rma.rmcr.get())
+    {
+        fillDerivedFields(*rma.rmcr, scp);
+    }
+    else if (rma.inca.get())
+    {
+        fillDerivedFields(*rma.inca, scp, arpPoly);
+    }
+}
+
+void Grid::fillDerivedFields(const RMAT& rmat, const Vector3& scp)
+{
+    // Row/Col.UnitVector and Derived fields
+    if (Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(rmat, scp);
+        col->unitVector = derivedColUnitVector(rmat, scp);
+    }
+}
+
+void Grid::fillDerivedFields(const RMCR& rmcr, const Vector3& scp)
+{
+    // Row/Col.UnitVector and Derived fields
+    if (Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(rmcr, scp);
+        col->unitVector = derivedColUnitVector(rmcr, scp);
+    }
+}
+
+void Grid::fillDerivedFields(const INCA& inca, const Vector3& scp,
+        const PolyXYZ& arpPoly)
+{
+
+    if (!Init::isUndefined<Poly1D>(inca.timeCAPoly) &&
+        !Init::isUndefined<PolyXYZ>(arpPoly) &&
+        Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(inca, scp, arpPoly);
+        col->unitVector = derivedColUnitVector(inca, scp, arpPoly);
+    }
+
+    if (Init::isUndefined<double>(col->kCenter))
+    {
+        col->kCenter = 0;
+    }
+
+    if (!Init::isUndefined<double>(inca.freqZero) &&
+        Init::isUndefined<double>(row->kCenter))
+    {
+        row->kCenter = derivedRowKCenter(inca);
+    }
+}
+
+void Grid::fillDerivedFields(const RgAzComp& rgAzComp,
+        const GeoData& geoData,
+        const SCPCOA& scpcoa,
+        double fc)
+{
+    const Vector3& scp = geoData.scp.ecf;
+
+    if (imagePlane == ComplexImagePlaneType::NOT_SET)
+    {
+        imagePlane = ComplexImagePlaneType::SLANT;
+    }
+    if (imagePlane == ComplexImageGridType::NOT_SET)
+    {
+        type = ComplexImageGridType::RGAZIM;
+    }
+
+    if (Init::isUndefined<Vector3>(row->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(scpcoa, scp);
+    }
+    if (Init::isUndefined<Vector3>(col->unitVector))
+    {
+        col->unitVector = derivedColUnitVector(scpcoa, scp);
+    }
+
+    row->fillDerivedFields(rgAzComp, geoData,
+            fc * 2 / math::Constants::SPEED_OF_LIGHT_METERS_PER_SEC);
```

Comment:
In the calling logic above, `fc` could be set to the reserved undefined value right?  So here it'll get set to some weird non-undefined value.  Does the Matlab code handle this any differently?